.github/workflows/build.yml

```
@@ -44,6 +44,11 @@ jobs:
       - name: Run pre-commit on all files
         run: pre-commit run --all-files
   test:
+    strategy:
+      matrix:
```

I think it would be helpful for all strategies to run even if one of them fails.

```suggestion
      fail-fast: false
      matrix:
```